src/main/java/com/google/sps/servlets/BlobUploadUrlServlet.java

```diff
@@ -0,0 +1,45 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * When the fetch() function requests the /blob-upload-url URL, the content of the response is
+ * the URL that allows a user to upload a file to Blobstore.
+ */
+@WebServlet("/blob-upload-url")
+public class BlobUploadUrlServlet extends HttpServlet {
+
+    private BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        setResponse(request, response, blobstoreService);
+    }
+
+    public void setResponse(HttpServletRequest request, HttpServletResponse response, BlobstoreService blobstoreService) throws IOException {
+        String uploadUrl = blobstoreService.createUploadUrl("/registration");
```

Comment:
Does this mean the user will always be redirected to registration.html when uploading a picture?